### DISCUSSION TITLE
What kernel size does the Lanczos filter use?

### DISCUSSION THREAD

**Discussion by mattpodolak** [python-pillow/Pillow#5519] [Q&A] [answered]
I am taking a look through the documentation and I don't see any mention of the kernel size used for the Lanczos filter for resizing. Does anyone have any insights into this?

**Comment by mattpodolak** [comment, marked as answer]
Looking at the C code it looks like it should be a kernel size of 3:
```c
lanczos_filter(double x) {    
/* truncated sinc */    
if (-3.0 <= x && x < 3.0) {        
return sinc_filter(x) * sinc_filter(x / 3);    
} 
return 0.0;
}
```

**Reply by radarhere** [reply]
Yes, I agree. The code you've found matches https://en.wikipedia.org/wiki/Lanczos_resampling#Lanczos_kernel, where 'a' is 3.
